### PR TITLE
Symfony clean after migration

### LINKED ISSUE #34596: Symfony 6 migration, tasks to do after actual upgrade
### Discussed in https://github.com/PrestaShop/PrestaShop/discussions/34594

<div type='discussions-op-text'>

<sup>Originally posted by **matthieu-rolland** November 16, 2023</sup>
### The problem

As we are progressing through the removal of deprecated elements for the Symfony 6 migration, we noticed that some things can't be changed until the very end, in the final Symfony upgrade PR.

### The solution

Here's the list (to be completed as we progress):

- Remove `UserProviderInterface::loadUserByUsername()` in favor of `UserProviderInterface::loadUserByIdentifier()`
- Remove `UserInterface::getUsername()` in favor of `UserInterface::getUserIdentifier()`
- The modification on `LegacyControllerContext` https://github.com/PrestaShop/PrestaShop/pull/34560 can be reverted (we can put back the readonly keyword as it should be handled correctly in Symfony 6.2)
- Remove `createAuthenticatedToken` function in `TokenAuthenticator`

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Symfony clean after migration
| Type?             | refacto
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| How to test?      | CI green and UI tests green
| UI Tests          | https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/9403098785
| Fixed issue or discussion?     | Fixes #34596
| Related PRs       | ~
| Sponsor company   | ~

### CODE REVIEW

**Review by matthieu-rolland — APPROVED** [review]

**Review by nicosomb — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @jolelievre ,

Install PR OK
Tests auto OK
Standards test OK

Thanks!

**Comment by ps-jarvis** [comment]
QA approved, well done! Message to the maintainers: do not forget to milestone it before the merge.

**Pull request merged**
